nautobot/extras/forms.py

```diff
@@ -437,6 +438,7 @@ class ConfigContextForm(BootstrapMixin, forms.ModelForm):
     regions = DynamicModelMultipleChoiceField(queryset=Region.objects.all(), required=False)
     sites = DynamicModelMultipleChoiceField(queryset=Site.objects.all(), required=False)
     roles = DynamicModelMultipleChoiceField(queryset=DeviceRole.objects.all(), required=False)
+    types = DynamicModelMultipleChoiceField(queryset=DeviceType.objects.all(), required=False)
```

Comment:
`types` is a fairly generic term. I can see why you picked it (parallel to `roles` for `DeviceRole`), but my personal preference would be to follow NetBox's example (https://github.com/netbox-community/netbox/commit/cd629fc7372432cdaecf072891f661bd4ad6d499) and call this field `device_types` just to reduce ambiguity - recall that config contexts can apply to virtual machines as well as devices.